### PR TITLE
[ESIMD] Add test for double GRF mode support.

### PR DESCRIPTION
**Pull request by kbobrovs**
Complementary compiler PR https://github.com/intel/llvm/pull/6182
Signed-off-by: Konstantin S Bobrovsky <konstantin.s.bobrovsky@intel.com>

### CODE REVIEW

File: SYCL/ESIMD/double_grf.cpp [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+// This test verifies effect of
+//   set_kernel_properties(kernel_properties::use_double_grf);
+// API call in device code:
+// - ESIMD/SYCL splitting happens as usual
+// - ESIMD module is further split into callgraphs for entry points requesting
+//   "double GRF" and callgraphs for entry points which are not
+// - ESIMD device binary images requesting "double GRF" must be compiled with
+//   -doubleGRF option
+
+// RxEQUIRES: gpu-intel-pvc
```
Comment (kbobrovs) [inline SYCL/ESIMD/double_grf.cpp]:
```suggestion
// REQUIRES: gpu-intel-pvc
```

File: SYCL/ESIMD/double_grf.cpp [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+// Kernels requesting larger GRF are grouped into separate module and compiled
+// with -doubleGRF regardless of SYCL_PROGRAM_COMPILE_OPTIONS value.
+
+// CHECK-LABEL: ---> piProgramBuild(
+// CHECK-NO-VAR: -vc-codegen -disable-finalizer-msg -doubleGRF
+// CHECK-WITH-VAR: -g -vc-codegen -disable-finalizer-msg -doubleGRF
+// CHECK: ) ---> pi_result : PI_SUCCESS
+// CHECK-LABEL: ---> piKernelCreate(
+// CHECK: <const char *>: {{.*}}EsimdKernel
```
Comment (kbobrovs) [inline SYCL/ESIMD/double_grf.cpp]:
```suggestion
// CHECK: <const char *>: {{.*}}EsimdKernelDoubleGRF
```

File: SYCL/ESIMD/double_grf.cpp [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    auto dev = q.get_device();
+    std::cout << "Running on " << dev.get_info<info::device::name>() << "\n";
+
+    auto e = q.submit([&](handler &cgh) {
+      auto PA = bufa.get_access<access::mode::read_write>(cgh);
+      cgh.parallel_for<class SyclKernel>(Size,
+                                         [=](id<1> i) { PA[i] = PA[i] + 1; });
+    });
+    e.wait();
+  } catch (cl::sycl::exception const &e) {
```
Comment (v-klochkov) [inline SYCL/ESIMD/double_grf.cpp:85]: minor/non-blocking comment:
```suggestion
  } catch (sycl::exception const &e) {
```

**Review by v-klochkov — APPROVED** [review]

**Comment by againull** [comment]
Test was merged in scope of https://github.com/intel/llvm-test-suite/pull/1686

**Pull request closed without merging**